### ISSUE TITLE
[BUG] check-signal-stop-start reports PASS even though I/O errors happened

### ISSUE THREAD

**Issue by miRoox** [thesofproject/sof-test#988]
**Describe the bug**
check-signal-stop-start reports PASS even though I/O errors happened

**To Reproduce**
```
TPLG=/lib/firmware/intel/sof-ace-tplg/sof-mtl-nocodec.tplg MODEL=MTLP_RVP_NOCODEC ~/sof-test/test-case/check-signal-stop-start.sh -c 10 -m playback
```

**Expected behavior**
check-signal-stop-start should report FAIL if I/O errors happened

**Detail Info**
1) Branch name and commit hash of the 3 repositories: sof (firmware/topology), linux (kernel driver) and sof-test (test case)
    * Kernel: https://github.com/thesofproject/linux/commit/fadccd8d7d21
    * SOF: https://github.com/thesofproject/sof/commit/24c0aad8328b
    * SOF-TEST: 3e021ab
2) Test report ID (if you find it from test report)
    * ID: 18469
3) Test DUT Model (or a brief discribtion about the device)
    * MODEL: MTLP_RVP_NOCODEC
4) Test TPLG
    * TPLG: sof-ace-tplg/sof-mtl-nocodec.tplg
5) Test case (what test script and how you run it)
    * TESTCASE: check-signal-stop-start.sh

**Screenshots or console output**
```
2022-12-01 22:30:44 UTC [REMOTE_INFO] ktime=1753 sof-test PID=5243: starting
2022-12-01 22:30:45 UTC [REMOTE_INFO] Starting /usr/local/bin/mtrace-reader.py
2022-12-01 22:30:45 UTC [REMOTE_INFO] /home/ubuntu/sof-test/test-case/check-signal-stop-start.sh will use topology /usr/lib/firmware/intel/sof-ace-tplg/sof-mtl-nocodec.tplg to run the test case
2022-12-01 22:30:45 UTC [REMOTE_INFO] Pipeline list to ignore is specified, will ignore 'pcm=HDA Digital' in test case
2022-12-01 22:30:45 UTC [REMOTE_INFO] Run command to get pipeline parameters
2022-12-01 22:30:45 UTC [REMOTE_COMMAND] sof-tplgreader.py /usr/lib/firmware/intel/sof-ace-tplg/sof-mtl-nocodec.tplg -f 'type:playback & ~pcm:HDMI & ~pcm:DMIC' -b ' pcm:HDA Digital' -s 0 -e
2022-12-01 22:30:45 UTC [REMOTE_INFO] Testing: run stop/start test on PCM:Port0,playback. Interval time: 0.5
2022-12-01 22:30:45 UTC [REMOTE_COMMAND] aplay -Dhw:0,0 -r 48000 -c 2 -f S16_LE /dev/zero -q &
2022-12-01 22:30:46 UTC [REMOTE_INFO] Stop/start count: 1
underrun!!! (at least 192.851 ms long)
2022-12-01 22:30:47 UTC [REMOTE_INFO] Stop/start count: 2
2022-12-01 22:30:48 UTC [REMOTE_INFO] Stop/start count: 3
aplay: pcm_write:2127: write error: Input/output error
2022-12-01 22:30:49 UTC [REMOTE_COMMAND] kill process: kill -9 5340
/home/ubuntu/sof-test/test-case/check-signal-stop-start.sh: line 116: kill: (5340) - No such process
2022-12-01 22:30:49 UTC [REMOTE_INFO] Testing: run stop/start test on PCM:Port1,playback. Interval time: 0.5
2022-12-01 22:30:49 UTC [REMOTE_COMMAND] aplay -Dhw:0,1 -r 48000 -c 2 -f S16_LE /dev/zero -q &
2022-12-01 22:30:50 UTC [REMOTE_INFO] Stop/start count: 1
underrun!!! (at least 168.151 ms long)
2022-12-01 22:30:51 UTC [REMOTE_INFO] Stop/start count: 2
2022-12-01 22:30:52 UTC [REMOTE_INFO] Stop/start count: 3
aplay: pcm_write:2127: write error: Input/output error
2022-12-01 22:30:53 UTC [REMOTE_COMMAND] kill process: kill -9 5381
/home/ubuntu/sof-test/test-case/check-signal-stop-start.sh: line 116: kill: (5381) - No such process
2022-12-01 22:30:53 UTC [REMOTE_INFO] Testing: run stop/start test on PCM:Port2,playback. Interval time: 0.5
2022-12-01 22:30:53 UTC [REMOTE_COMMAND] aplay -Dhw:0,2 -r 48000 -c 2 -f S16_LE /dev/zero -q &
2022-12-01 22:30:53 UTC [REMOTE_INFO] Stop/start count: 1
underrun!!! (at least 163.285 ms long)
2022-12-01 22:30:54 UTC [REMOTE_INFO] Stop/start count: 2
2022-12-01 22:30:55 UTC [REMOTE_INFO] Stop/start count: 3
aplay: pcm_write:2127: write error: Input/output error
2022-12-01 22:30:56 UTC [REMOTE_COMMAND] kill process: kill -9 5419
/home/ubuntu/sof-test/test-case/check-signal-stop-start.sh: line 116: kill: (5419) - No such process
declare -- cmd="journalctl_cmd --since=@1669933837"
2022-12-01 22:30:57 UTC [REMOTE_INFO] pkill -TERM -f mtrace-reader.py
2022-12-01 22:30:57 UTC [REMOTE_INFO] ktime=1766 sof-test PID=5243: ending
2022-12-01 22:30:57 UTC [REMOTE_INFO] Test Result: PASS!
```

**Comment by fredoh9** [comment]
I can easily reproduce this problem. Will submit a fix soon

**Comment by miRoox** [comment]
Fixed by #988
